src/main/java/jenkins/branch/NoTriggerBranchProperty.java

```diff
@@ -95,7 +125,40 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
             }
             return true;
         }
-
     }
 
+    /**
+     * Strategy which determines which builds should be suppressed.
+     * @since 2.7.1
+     */
+    public enum SuppressionStrategy {
+        /**
+         * All builds triggered by SCM are suppressed.
+         *
+         * @since 2.7.1
+         */
+        ALL(Messages._NoTriggerBranchProperty_strategy_all()),
+        /**
+         * Only builds triggered by {@link BranchIndexingCause} are suppressed.
+         *
+         * @since 2.7.1
+         */
+        INDEXING(Messages._NoTriggerBranchProperty_strategy_indexing()),
+        /**
+         * Only builds triggered by {@link BranchEventCause} are suppressed.
+         *
+         * @since 2.7.1
```

Comment:
NIT: same comment as above - or just remove this as it is the same as the enum type.